src/main/java/com/squareup/kotlinpoet/PropertySpec.kt

```diff
@@ -149,12 +156,22 @@ class PropertySpec private constructor(builder: Builder) {
       require(getter.name == GETTER) { "${getter.name} is not a getter" }
       check(this.getter == null) { "getter was already set" }
       this.getter = getter
+      checkPromoteInlineToProperty()
     }
 
     fun setter(setter: FunSpec) = apply {
       require(setter.name == SETTER) { "${setter.name} is not a setter" }
       check(this.setter == null) { "setter was already set" }
       this.setter = setter
+      checkPromoteInlineToProperty()
+    }
+
+    private fun checkPromoteInlineToProperty() {
```

Comment:
Can you do this at render time instead?

Reply:
Ie. keep `modifiers` as-is as a property of this class, but add `inline` when printing it? That way if anyone uses KotlinPoet as a basic AST they get `inline` where they expect it

Reply:
And (more importantly in my opinion) they don't see `inline` where they didn't specify it.

Reply:
Agree, good point. Force-pushed an updated version.